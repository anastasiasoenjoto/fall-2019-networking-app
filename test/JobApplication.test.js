import React from 'react'
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import JobApplication from '../frontend/src/components/JobApplicationForm'




describe('Job Application', () => {
  
  it('renders correctly', () => {
    const wrapper = shallow(<JobApplication />);
    expect(wrapper).toMatchSnapshot();
  });

  it('test for onChangeName', () => {
    const event = {
      target: { 
        value: 'firstNameTest'
      }
    };
    const wrapper = shallow(
      <JobApplication />);
    
    wrapper.find('#nameOfApplicant').props().onChange(event);
    expect(wrapper.state('nameOfApplicant')).toEqual('firstNameTest');
  });

  it('test for onChangePhone', () => {
    const event = {
      target: { value: 'phoneNumb'}
    }
    const wrapper = shallow(
      <JobApplication/>
    );
    wrapper.find('#phone').props().onChange(event);
    expect(wrapper.state('PhoneNumber')).toEqual('phoneNumb');


  });

  it('test for onChangeEmail', () => {
    const event = {
      target: { value: 'emailTest'}
    }
    const wrapper = shallow(
      <JobApplication />
    );
    wrapper.find('#email').props().onChange(event);
    expect(wrapper.state('email')).toEqual('emailTest');


  });

  it('test for onChangeSkills', () => {
    const event = {
      target: { value: 'skillsTest'}
    }
    const wrapper = shallow(
      <JobApplication />
    );
    wrapper.find('#skill').props().onChange(event);
    expect(wrapper.state('skill')).toEqual('skillsTest');


  });

  it('test for onChangeResume', () => {
    const event = {
      target: { value: 'testURL'}
    }
    
    const wrapper = mount(
      <JobApplication/>
    );
    wrapper.find('#resume').props().onChange(event);
    expect(wrapper.state('resume')).toEqual('testURL');
  });

  it('test for onChangeMajor', () => {
    const event = {
      target: { value: 'anthropology'}
    }

    const wrapper = mount(
      <JobApplication/>
    );
    wrapper.find('#major').props().onChange(event);
    expect(wrapper.state('major')).toEqual('anthropology');
  });

  it('test for onChangeGPA', () => {
    const event = {
      target: { value: '3.54'}
    }
    const wrapper = shallow(
      <JobApplication/>,
    );
    wrapper.find('#GPA').props().onChange(event);
    expect(wrapper.state('GPA')).toEqual('3.54');


  });
})