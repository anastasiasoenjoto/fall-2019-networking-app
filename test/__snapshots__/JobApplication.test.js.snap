// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Application renders correctly 1`] = `ShallowWrapper {}`;
